src/utils/flatten.js

```diff
@@ -52,6 +53,20 @@ const flatten = (
       return ruleSet
     }
 
+    /* Throw if a React Element was given styles */
+    if (React.isValidElement(chunk)) {
+      const elementName =
+        /* $FlowFixMe TODO: flow for chunk of type React Element. Not sure if Element should be added to Interpolation */
+        typeof chunk.type === 'string'
+          ? chunk.type
+          : chunk.type.displayName || chunk.type.name
+      throw new StyledError(
+        1,
+        `A plain React class (${elementName}) 
```

Comment:
This is a little inaccurate :) it says “class” here instead of “element “

I’d also either remove “your styles” to “has been interpolated into styles” or just change it to an active voice (“ You have interpolated a React element into your styles”)

Reply:
Yeah I don't want it to occur so late. We want to catch the original class being interpolated so the resulting error is useful

Reply:
True. I felt the same way about catching the error sooner. Let me know if you know a good place (file/line number) for me to put this. I will see if I can find it out myself.

Reply:
Done. Please take a look.